### PR TITLE
convenience function to help compare models using WAIC or LOO

### PR DESCRIPTION
**Pull request by aloctavodia**
This is a simple function that computes WAIC or LOO (IC) for a set of models and return a DataFrame ordered from "best" to "worse" model (in terms of the IC). It also computes other quantities like Akaike weights that can be used for example for model averaging and the standard error of the difference in IC between each model and the top-ranked model, that can help to compare and interpret IC for different models. A similar function is used extensively in the book Statistical Rethinking by Richard McElreath.

In a future PR I will add a plot based on this function (I guess it should be a separate function living inside `plots`). And I will update the notebook examples to give some details on this function and how it can helps to interpret IC (beyond the "smaller is better" rule ).

One current problem with this function is that the warnings returned by waic/loo are not very useful given that the user does not know which model is returning a warning or even if there is more than one warning.

I am not sure which is the best solution here. A couple of alternatives are:
1. To catch the warning and print it together with the model index.
2. To add a "warning" column in the output DataFrame (filled with boolean maybe). These values could be obtained by catching the warning inside `compare`, or by modifying waic/loo to return such a value.

I think the second option (including the modification of waic/loo) is a better option. What do you think?

### CODE REVIEW

**Comment by twiecki** [comment]
@aloctavodia Catching the warnings would be the pythonic way I think.

File: pymc3/stats.py [hunk 1]
```diff
@@ -281,10 +281,10 @@ def compare(traces, models, ic='WAIC'):
     Returns
     -------
     A DataFrame, ordered from lowest to highest IC. The index reflects
-    the order in which the models are passed to this function. The columns are:
+    the order in wich the models are passed to this function. The columns are:
```
Comment (twiecki) [inline pymc3/stats.py]: typo

File: pymc3/stats.py [hunk 2]
```diff
@@ -281,10 +281,10 @@ def compare(traces, models, ic='WAIC'):
...
     IC: Information Criteria (WAIC or LOO).
-        Smaller IC indicates higher out-of-sample predictive fit ("better" model).
-        Default WAIC.
+        Smaller IC indicates higher out-of-sample predictive fit ("better" model). 
+        Default WAIC.
```
Comment (twiecki) [inline pymc3/stats.py]: stray spaces

**Comment by twiecki** [comment]
An example would also be good.

File: pymc3/stats.py [hunk 3]
```diff
@@ -262,6 +262,90 @@ def bpic(trace, model=None):
...
+def compare(traces, models, ic='WAIC'):
+    """
+    Compare models based on the widely available information criterion (WAIC)
+    or leave-one-out (LOO) cross-validation.
+    Read more theory here - in a paper by some of the leading authorities on
+    Model Selection - dx.doi.org/10.1111/1467-9868.00353
+
+    Parameters
+    ----------
+    traces: list of PyMC3 traces
```
Comment (twiecki) [inline pymc3/stats.py]: `varname : description`

**Comment by aloctavodia** [comment]
@twiecki thanks for the reviews!

**Pull request merged**